authz/openfga/collaboration_model/v1/schema.fga

```diff
@@ -0,0 +1,214 @@
+model
+  schema 1.1
+
+type user
+
+type role
+  relations
+    define assignee: [user, role#assignee]
+
+    # ------------------ Actions ------------------
+    define can_assume: assignee
+
+type server
+  relations
+    # Relation Hierarchies
+    define child: [project]
+
+    # ------------------ Built-in Roles ------------------
+    define global_admin: [user, role#assignee]
+
+    # ------------------ Actions ------------------
+    define can_create_project: global_admin
+
+    # ------------------ Assignable Privileges ------------------
+    # Anyone can list projects, they are however filtered by can_show_in_list of the project.
+    # Only global_admin can list all projects.
+    define can_list_all_projects: global_admin
+
+    # GRANT Permissions
+    define can_grant_global_admin: global_admin
+
+type project
+  relations
+    # Relation Hierarchies
+    define child: [warehouse]
+    define parent: [server]
+
+    # ------------------ Built-in Roles ------------------
+    # Project Admin has two purposes:
+    # 1. Lock-out protection: Checked to never be empty
+    # 2. Granting new privileges, such as billing, which might require separate permissions / admin roles
+    define project_admin: [user, role#assignee] or global_admin from parent
+    # Security Admins can manage all security aspects (grants, ownership) but not modify, create or access objects
+    define security_admin: [user, role#assignee] or project_admin
+    # Warehouse Admins can manage all warehouse aspects but not grant privileges
+    define warehouse_admin: [user, role#assignee] or project_admin
+
+    # ------------------ Assignable Privileges ------------------
+    # Privileges relevant for the warehouse and sub-objects
+    define describe: [role#assignee] or select or create or warehouse_admin or security_admin
+    define select: [role#assignee] or modify
+    define create: [role#assignee] or warehouse_admin
+    define modify: [role#assignee] or project_admin
+
+    # ------------------ Actions ------------------
+    define can_create_warehouse: create  # Create stuff inside the project / create new warehouses
+    define can_delete: modify  # Drop this project
+    define can_get_metadata: describe or can_get_metadata from child
+    define can_list_warehouses: can_get_metadata # Can list elements in this projects - will be filtered subsequently
+    define can_show_in_list: can_get_metadata
```

Comment:
Maybe I got it wrong, but this falls a bit out of the pattern? the others can be read as:

`Actor can do action on project`, this one is more like `Project can show in list`, maybe `can_include_in_list`?

Reply:
Renamed